### DISCUSSION TITLE
Tmds.DBus.DBusException on ubuntu22.04.1?

### DISCUSSION THREAD

**Discussion by BonanzaBegin** [AvaloniaUI/Avalonia#9747] [Q&A]
022-12-20 19:00:35 673 [Error]:Exception type:Tmds.DBus.DBusException,Exception message:org.freedesktop.DBus.Error.UnknownMethod: No such method “Destroy”,Exception from:Tmds.DBus,Exception StackTrace:   at Tmds.DBus.DBusConnection.CallMethodAsync(Message msg, Boolean checkConnected, Boolean checkReplyType)
   at Tmds.DBus.Connection.CallMethodAsync(Message message)
   at Tmds.DBus.CodeGen.DBusObjectProxy.SendMethodReturnReaderAsync(String iface, String member, Nullable`1 inSignature, MessageWriter writer)

when my Avalonia app run on ubuntu22.04.1,the exception that happens frequently,although it doesnt to crash my app and i can inogre it ,i want to know how i can avoid the exception ,any one can help?

**Comment by kekekeks** [comment]
It looks like that a new method was added to some DBus interface and we don't implement it. No idea which one though.

**Reply by BonanzaBegin** [reply]
Method name is Destroy?   it will be called frequently,such as while the popup in combobox cloing,window size changing and so on.......
At this moment, I completely ignore such exceptions by this way,

TaskScheduler.UnobservedTaskException += (sender, e) =>
            {
                var exs = e.Exception.InnerExceptions.Where(e => e.GetType() != typeof(Tmds.DBus.DBusException));
                if (exs.Count() > 0)
                {
                    string errorMsg = string.Join("\r\n", exs.Select(ex =>
                            $"Exception type:{ex.GetType().FullName}," +
                            $"Exception message:{ex.Message},Exception from:{ex.Source},Exception StackTrace:" +
                            $"{ex.StackTrace}"));
                    app._Logger?.Error(errorMsg);
                }

                e.SetObserved();
            };

But it doesn't feel perfect...

**Comment by kekekeks** [comment]
Enabling "Stop on all exceptions" and examining the `iface` value in `Tmds.DBus.CodeGen.DBusObjectProxy.SendMethodReturnReaderAsync` could help to determine the dbus interface

**Reply by BonanzaBegin** [reply]
I don't know how to do it ,could you please show me some code? thank you
